src/libraries/System.Private.CoreLib/src/System/IndexOfAnyValues/ProbabilisticMap.cs

```diff
@@ -23,8 +26,19 @@ namespace System.Buffers
     [StructLayout(LayoutKind.Sequential)]
     internal readonly struct ProbabilisticMap
     {
-        private const int IndexMask = 0x7;
-        private const int IndexShift = 0x3;
+        // We need at least Sse41 for the hardware accelerated ConditionalSelect support.
+        private static bool IsVectorizationSupported => Sse41.IsSupported || AdvSimd.Arm64.IsSupported;
```

Comment:
This is not `readonly` which means it incurs an actual check per use.

Making it a `static readonly` is also strictly worse sicne these are otherwise treated as JIT time constants (always, even for debug code).

Reply:
Ah, I misread. It's a property.

Still believe you should keep these as direct usages so the normal benefits of these being treated as constants kicks in.

Reply:
> This is not readonly which means it incurs an actual check per use.

This won't get inlined and be equivalent to if you wrote `if (Sse41.IsSupported || AdvSimd.Arm64.IsSupported)` at the call site?

Reply:
Nevermind, my question overlapped with your revision.

Reply:
It's a property and not a field, so should be treated as JIT constant.

Reply:
I can move the check inline, though it's a bit unfortunate to have to duplicate it.

Btw I've used the same pattern (property) in IndexOfAnyAsciiSearcher and PackedSpanHelpers already. Do you think those should also be updated? It'd be great if the JIT could just treat these as equivalent so that wasn't necessary ofc.

Reply:
> It'd be great if the JIT could just treat these as equivalent so that wasn't necessary ofc.

Right, the general issue is in the cost required to resolve non-intrinsic methods, scan the IL (which then requires further resolution of methods - this time intrinsic), and then make a decision around whether inlining is profitable or not.

There are many factors that can make this fail or be less than optimal and its a balance between what's good for typical user code and what's good for highly optimized hot code.

> Btw I've used the same pattern (property) in IndexOfAnyAsciiSearcher and PackedSpanHelpers already

There is likely some benefit to doing so, long term. It's also likely not the "worst thing" for these methods and the JIT is probably doing the "right thing" in Tier 1. It's just having to do more to make it happen

Reply:
In the case of intrinsic methods, the VM flags them specially and so resolution tends to be a lot cheaper and we can make an easy determination of "this is more than likely profitable".

We can also avoid the inliner entirely by creating new IR nodes directly. In the case of `Isa.IsSupported`, this always produces a constant `true` or constant `false` node and so its as if the user had directly written `if (true)` or `if (false)` themselves.

Reply:
Moved the checks inline in this file